inductiva/resources/machine_groups.py

```diff
@@ -0,0 +1,106 @@
+    """Returns a string representation of a list of machine groups."""
+    columns = [
+        "Name", "VM Type", "# machines", "Disk Size in GB", "Spot", "Created at"
+    ]
+    rows = []
+
+    for machine_group in machine_groups:
+        rows.append([
+            machine_group["name"], machine_group["machine_type"],
+            machine_group["num_instances"], machine_group["disk_size_gb"],
+            bool(machine_group["spot"]), machine_group["create_time"]
+        ])
+
+    formatters = {"Created at": format_utils.datetime_formatter}
+    override_col_space = {
+        "VM Type": 15,
+        "# machines": 12,
+        "Spot": 10,
+    }
+
+    return format_utils.get_tabular_str(
+        rows,
+        columns,
+        default_col_space=18,
+        override_col_space=override_col_space,
+        formatters=formatters,
+    )
+
+
+def _fetch_machine_groups_from_api():
+    """Get all active machine groups of a user from the API."""
+    try:
+        api = instance_api.InstanceApi(inductiva.api.get_client())
+        response = api.list_active_user_instance_groups()
+        if len(response.body) == 0:
+            logging.info("No active machine groups found.")
+            return response.body
+
+        return response.body
+
+    except inductiva.client.ApiException as api_exception:
+        raise api_exception
+
+
+# pylint: disable=redefined-builtin
+def list():
+    # pylint: disable=line-too-long
+    """Lists all active machine groups info.
+
+    This method queries Google Cloud for active machine groups
+    that belong to the user. It outputs all the information relative to
+    each machine group as folllows:
+
+    INFO:absl:Active machine groups:
+                                        Name         VM Type   # machines    Disk Size in GB       Spot         Created at
+    api-1b1f724c-5cfe-4d87-8439-9689aa139723   c2-standard-4            1                 40      False   13 Sep, 07:38:50
+    api-8e6bf7d8-4888-4de9-bda5-268484b46e6f   c2-standard-4            1                 40      False   13 Sep, 07:37:49
+
+    The name of the machine group can be used to retrieve a MachineGroup object
+    with the 'get' function."""
+    # pylint: enable=line-too-long
+
+    machine_groups = _fetch_machine_groups_from_api()
+    print("Active machine groups:")
+    print(_machine_group_list_to_str(machine_groups))
+
+
+def get(name: str):
+    """Returns a 'MachineGroup' object.
+
+    Given the name of the machine group, returns a 'MachineGroup' object
+    with the attributes of that machine group.
+
+    Args:
+        name: Name of the machine group."""
+    try:
+        api = instance_api.InstanceApi(inductiva.api.get_client())
+        response = api.get_instance_group({"name": name})
+
+        return resources.MachineGroup.from_api_response(response.body)
+
+    except inductiva.client.ApiException as api_exception:
+        raise api_exception
+
+
+def get_all():
```

Comment:
I think that in the task we don't have get and get_all, and also: list is listing all but is not called "list_all" . I think it would be enough to have a get method that return a list with all the  machine group objects, like we did in tasks, no?

Reply:
I am not sure about it, but I can comment out get() function for now and see based on the use cases  what type of functions make the most sense and are useful, wdyt?